pkg/process/handler.go

```diff
@@ -2,34 +2,175 @@ package process
 
 import (
 	"github.com/AsterNighT/software-engineering-backend/api"
-	"github.com/labstack/echo/v4"
 	"net/http"
 )
 
 type RegistrationHandler struct {
 }
 
 // @Summary get all departments
-// @Tags departments
+// @Tags patient
 // @Description display all departments of a hospital
 // @Produce json
-// @Success 200 {object} api.ReturnedData{data=[]Department}
-// @Router /departments [GET]
+// @Success 200 {array} api.ReturnedData{data=[]Department}
+// @Router /patient/departments [GET]
 func (h *RegistrationHandler) GetAllDepartments(c echo.Context) error {
 
 	c.Logger().Debug("hello world")
 	return c.JSON(http.StatusCreated, api.Return("pong", nil))
 }
 
 // @Summary get a department by its ID
-// @Tags departments
+// @Tags patient
 // @Description return a department's details by its ID
 // @Param department_id path uint true "Department ID"
 // @Produce json
 // @Success 200 {object} api.ReturnedData{data=Department}
-// @Router /departments/{department_id} [GET]
+// @Router /patient/departments/{department_id} [GET]
 func (h *RegistrationHandler) GetDepartmentByID(c echo.Context) error {
 
 	c.Logger().Debug("hello world")
 	return c.JSON(http.StatusCreated, api.Return("pong", nil))
 }
+
+// @Summary patient register
+// @Tags patient
+// @Description return registration state
+// @Param patient_id, department_id
```

Comment:
where are you putting them? I suggest `patientID` be in path and `departmentID` be in body

Reply:
I think if we have more than one param, we had better use POST method and put all params in request body